iocore/eventsystem/UnixEventProcessor.cc

```diff
@@ -152,6 +155,59 @@ EventProcessor::start(int n_event_threads, size_t stacksize)
 #else
       Debug("iocore_thread", "EThread: %d %s: %d", i, obj_name, obj->logical_index);
 #endif // HWLOC_API_VERSION
+    }
+#endif // TS_USE_HWLOC
+
+    snprintf(thr_name, MAX_THREAD_NAME_LENGTH, "[ET_NET %d]", i);
+#if TS_USE_HWLOC
+    if (obj_count > 0) {
+      hwloc_membind_policy_t mem_policy = HWLOC_MEMBIND_DEFAULT;
+      hwloc_nodeset_t nodeset           = hwloc_bitmap_alloc();
+      int num_nodes                     = 0;
+
+      hwloc_cpuset_to_nodeset(ink_get_topology(), obj->cpuset, nodeset);
+      num_nodes = hwloc_get_nbobjs_inside_cpuset_by_type(ink_get_topology(), obj->cpuset, HWLOC_OBJ_NODE);
+
+      if (num_nodes == 1) {
+        mem_policy = HWLOC_MEMBIND_BIND;
+      } else if (num_nodes > 1) {
+        mem_policy = HWLOC_MEMBIND_INTERLEAVE;
+      }
+
+      if (mem_policy != HWLOC_MEMBIND_DEFAULT) {
+        hwloc_set_membind_nodeset(ink_get_topology(), nodeset, mem_policy, HWLOC_MEMBIND_THREAD);
+      }
```

Comment:
Commenting this would be really helpful. I'd also recommend breaking it into helper functions so that it can be understood without necessarily having the ``hwloc`` man page in hand.